docs/docs/adrs/adr-013-equivocation-slashing.md

```diff
@@ -0,0 +1,169 @@
+
+### Implementation
+The aggressive approach allows for multiple simplifications. Specifically, we intend to introduce the following logic in the [HandleConsumerDoubleVoting](https://github.com/cosmos/interchain-security/pull/1232) method:
+```go
+undelegationsTotalAmount := sdk.NewInt(0)
+for _, v := range k.stakingKeeper.GetUnbondingDelegationsFromValidator(ctx, validatorAddress) {
+    for _, entry := range v.Entries {
+        undelegationsTotalAmount = undelegationsTotalAmount.Add(entry.InitialBalance)
+    }
+}
+
+redelegationsTotalAmount := sdk.NewInt(0)
+for _, v := range k.stakingKeeper.GetRedelegationsFromSrcValidator(ctx, validatorAddress) {
+    for _, entry := range v.Entries {
+        redelegationsTotalAmount = redelegationsTotalAmount.Add(entry.InitialBalance)
+    }
+}
+
+infractionHeight := 0
+totalPower := validators voting Power + undelegationsTotalAmount.Add(redelegationsTotalAmount)
+slashFraction := k.slashingKeeper.SlashFractionDoubleSign(ctx)
+
+k.stakingKeeper.Slash(ctx, validatorConsAddress, infractionHeight, totalPower, slashFraction, DoubleSign)
+```
+
+**Infraction height:** We provide a zero `infractionHeight` to the [Slash](https://github.com/cosmos/cosmos-sdk/blob/v0.47.0/x/staking/keeper/slash.go#L33) method in order to slash all ongoing undelegations and redelegations (see checks in [Slash](https://github.com/cosmos/cosmos-sdk/blob/v0.47.0/x/staking/keeper/slash.go#L92), [SlashUnbondingDelegation](https://github.com/cosmos/cosmos-sdk/blob/v0.47.0/x/staking/keeper/slash.go#L195), and [SlashRedelegation](https://github.com/cosmos/cosmos-sdk/blob/v0.47.0/x/staking/keeper/slash.go#L249)).
+
+**Power:** We pass the sum of the voting power of the misbehaving validator and the power of all the ongoing undelegations and redelegations. This is a slightly more aggressive approach than just providing the voting `power` at the evidence height. If we assume that the `slashFactor` is 5%, then the `power` we pass is `validatorPower + totalPower(undelegations) + totalPower(redelegations)`.
+Hence, when the `Slash` method slashes all the undelegations and redelegations it would end up with `0.05 * validatorPower + 0.05 * totalPower(undelegations) + 0.05 * totalPower(redelegations) - 0.05 * totalPower(undelegations) - 0.05 * totalPower(redelegations) = 0.05 * validatorPower` and hence it would slash 5% of the `validatorPower` at evidence height.
+
+**Storing evidence:** As mentioned, we do not expire evidence with this aggressive approach. However, because we do not want to slash the same validator more than once for the same infraction, we have to store the evidence in some _cache_ in order to check if we have already slashed that validator. We intend to do the following:
```

Comment:
We don't need to store evidence. Validators that are slashed for double signing are tombstoned. Tombstoned validators cannot be slashed again.

Reply:
Yes, missed this. 🤦 
Removed the section altogether and added the following sentence:
>Note that we do not need to store equivocation evidence to avoid slashing a validator more than once, because we [do not slash](https://github.com/cosmos/cosmos-sdk/blob/v0.47.0/x/evidence/keeper/infraction.go#L94) tombstoned validators and we [tombstone](https://github.com/cosmos/cosmos-sdk/blob/v0.47.0/x/evidence/keeper/infraction.go#L138) a validator when slashed.